### PR TITLE
Collateral tokens usd price in vault details pages

### LINKED ISSUE #1594: `vault details` - add dollar value
#### What would you like to be added:
Add dollar value on token amounts
It should look like similar to the Edit Collateral page. Ex. **200 DFI/$10000**

1. Vault Detail - Collateral Tab
image
2. Loans tab - in Outstanding balance and interest amount
image

### LINKED ISSUE #1595: `add/remove collateral` - dollar value
#### What would you like to be added:
Display dollar value for Available and current fields
image

### LINKED ISSUE #1639: vault details - add dollar value for loans card
#### What would you like to be added:
image

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #1594 
Fixes #1595
Fixes #1639

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Loans/VaultDetail/components/CollateralsTab.tsx [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+  totalCollateralValue: BigNumber
 }
 
 export function CollateralsTab ({ vault }: {vault: LoanVault}): JSX.Element {
+  const collateralTokens = useSelector((state: RootState) => state.loans.collateralTokens)
+  const dispatch = useDispatch()
+  const client = useWhaleApiClient()
+
+  useEffect(() => {
+    dispatch(fetchCollateralTokens({ client }))
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Loans/VaultDetail/components/CollateralsTab.tsx:30]: Do we need to add this here? We're already fetching it on the first page

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Review by thedoublejay — CHANGES_REQUESTED** [review]
Screenshot_20211209-085653_Expo Go

File: mobile-app/app/screens/AppNavigator/screens/Loans/components/AddOrRemoveCollateralForm.tsx [hunk 2]
```diff
@@ -157,17 +159,48 @@ export const AddOrRemoveCollateralForm = React.memo(({ route }: Props): JSX.Elem
...
+                value={activePrice.multipliedBy(available).toFixed(2)}
+                thousandSeparator
+                decimalScale={2}
+                displayType='text'
+                prefix='$'
+                renderText={(val: string) => (
+                  <ThemedText
+                    dark={tailwind('text-gray-400')}
+                    light={tailwind('text-gray-500')}
+                    style={tailwind('text-xs')}
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Loans/components/AddOrRemoveCollateralForm.tsx:189]: Fix for alignment in android:
```suggestion
                    style={tailwind('text-xs leading-5')}
```

File: mobile-app/app/screens/AppNavigator/screens/Loans/components/AddOrRemoveCollateralForm.tsx [hunk 3]
```diff
@@ -157,17 +159,48 @@ export const AddOrRemoveCollateralForm = React.memo(({ route }: Props): JSX.Elem
...
         styleProps={tailwind('font-medium')}
-      />
-      {isConversionRequired &&
+      >
+        <ThemedText
+          light={tailwind('text-gray-700')}
+          dark={tailwind('text-gray-200')}
+          style={tailwind('text-sm font-medium')}
+        >
+          <Text>{' '}</Text>
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Loans/components/AddOrRemoveCollateralForm.tsx:175]: Can we use margin here?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Review by thedoublejay — DISMISSED** [review]

**Review by kyleleow — DISMISSED** [review]

**Review by thedoublejay — APPROVED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**